framework/auth/core.py

```diff
@@ -843,6 +848,22 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def confirm_token(self, token):
```

Comment:
Actually, there seems to already be a method that implements this behavior, `get_confirmation_token`. It will raise an `ExpiredTokenError` if the token is expired, and a KeyError if the token doesn't exist. Can you use that instead?